### PR TITLE
feat: added a splashscreen lab

### PR DESCRIPTION
**Pull request by simonhyll**
#### What kind of changes does this PR include?

- New or updated content

#### Description
Adds a new splashscreen lab.

### CODE REVIEW

**Comment by simonhyll** [comment]
It's not perfect yet but it's close enough now where I'd like some input on it. 😄

**Comment by vasfvitor** [comment]
I'm just looking at it right now lol, good timing

**Comment by vasfvitor** [comment]
what do you think of adding the full CLI as a way to quick start?
I guess it should be
> pnpm create tauri-app --beta --template vanilla --manager pnpm --mobile yes splashscreen-lab

**Comment by vasfvitor** [comment]
I'm gonna follow the lab and give my feedback based on it

File: src/content/docs/learn/splashscreen.mdx [hunk 1]
```diff
@@ -1,11 +1,253 @@
...
+    // Effectively a JavaScript main function
+    window.addEventListener("DOMContentLoaded", () => {
+        setup()
+    });
+    ```
+
+    ```rust
+    // /src-tauri/src/lib.rs
+    // Import functionalities we'll be using
+    use tauri::async_runtime::spawn;
```
Comment (vasfvitor) [inline src/content/docs/learn/splashscreen.mdx:161]:
```suggestion
    use tauri::{async_runtime::spawn, State, AppHandle};
```

File: src/content/docs/learn/splashscreen.mdx [hunk 2]
```diff
@@ -1,11 +1,253 @@
...
+        }
+        Ok(())
+    }
+
+    // An async function that does some heavy setup task
+    async fn setup(app: AppHandle) -> Result<(), ()> {
+        // Fake performing some heavy action for 3 seconds
+        println!("Performing really heavy backend setup task...");
+        sleep(Duration::from_secs(3)).await;
+        println!("Backend setup task completed!")
```
Comment (vasfvitor) [inline src/content/docs/learn/splashscreen.mdx]:
```suggestion
        println!("Backend setup task completed!");
```

File: src/content/docs/learn/splashscreen.mdx [hunk 3]
```diff
@@ -1,11 +1,253 @@
...
+        // setup hook instead!
+        tauri::Builder::default()
+            // Register a `State` to be managed by Tauri
+            // We need write access to it so we wrap it in a `Mutex`
+            .manage(Mutex::new(SetupState{
+                frontend_task: false, 
+                backend_task: false
+            }))
+            // Add a command we can use to check
+            .invoke_handler(tauri::generate_handler!([check_completion]))
```
Comment (vasfvitor) [inline src/content/docs/learn/splashscreen.mdx]:
```suggestion
            .invoke_handler(tauri::generate_handler![set_complete])
```

**Review by vasfvitor — CHANGES_REQUESTED** [review]
well, even after the requested changed it doesn't work. Tried to fix it, broke even more. I could give another shot, or do you want to do it yourself?

Is the goal to teach about state? Because if not, I think, if possible, it should be done without it. Or else, add a requirement lab that teaches about managing state. Speaking of which, I've read many times your webpage about state and it helped me A LOT when I needed it , really!

File: src/content/docs/learn/splashscreen.mdx (hunk 3 above)
Comment (vasfvitor) [reply src/content/docs/learn/splashscreen.mdx]: or the function could be renamted to check_completion

File: src/content/docs/learn/splashscreen.mdx [hunk 4]
```diff
@@ -1,11 +1,253 @@
...
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>Tauri App</title>
+    </head>
+    <body>
+        <div class="container">
+            <h1>Tauri used Splash!</h1>
+            <div class="row">
+                <h5>It was super effective!</h5>
+            </div>
+    </body>
```
Comment (dreyfus92) [inline src/content/docs/learn/splashscreen.mdx]: I think that you missed adding a closing tag for the parent `div` here.
```suggestion
            </div>
        </div>
    </body>
```

**Review by dreyfus92 — COMMENTED** [review]
Awesome job @simonhyll, I left just a tiny suggestion 🫡

File: src/content/docs/learn/splashscreen.mdx (hunk 4 above)
Comment (simonhyll) [reply src/content/docs/learn/splashscreen.mdx]: That I did indeed 😅

**Comment by simonhyll** [comment]
> what do you think of adding the full CLI as a way to quick start? I guess it should be
> 
> > pnpm create tauri-app --beta --template vanilla --manager pnpm --mobile yes splashscreen-lab

@vasfvitor The main reason I didn't do that was because it's very convenient to just add the CTA fragment we have to things and keep that as the main for everyone. Giving the full command feels less real-world adapted, the lab aims to provide some practical experience after all, might as well make people use the method they'd use the most in the future right? And if we do a full command we'd need to provide it for all package managers.

**Comment by simonhyll** [comment]
> well, even after the requested changed it doesn't work. Tried to fix it, broke even more. I could give another shot, or do you want to do it yourself?
> 
> Is the goal to teach about state? Because if not, I think, if possible, it should be done without it. Or else, add a requirement lab that teaches about managing state. Speaking of which, I've read many times your webpage about state and it helped me A LOT when I needed it , really!

Glad to head the article is of use to people ❤️

It's not explicitly the intent to teach about states but I also have a hard time seeing how to tackle this without states, or rather I have a hard time seeing a more appropriate solution to this specific use-case. We can definitely add a reference link to where people can read up more on states, not sure I'd say it's a prerequisite necessarily, I think I can include a very short text on what states are in the lab and it'll be good enough for the sort of copy-paste nature of labs

File: src/content/docs/learn/splashscreen.mdx (hunk 1 above)
Comment (simonhyll) [reply src/content/docs/learn/splashscreen.mdx:161]: Added, just slightly differently.

File: src/content/docs/learn/splashscreen.mdx (hunk 2 above)
Comment (simonhyll) [reply src/content/docs/learn/splashscreen.mdx]: Fixed

File: src/content/docs/learn/splashscreen.mdx (hunk 3 above)
Comment (simonhyll) [reply src/content/docs/learn/splashscreen.mdx]: Fixed

**Review by dreyfus92 — APPROVED** [review]
LGTM 😁

**Review by vasfvitor — APPROVED** [review]
I was going to suggest adding `"decorations": false`   to make it feel more splashyscreeney, but it just looks bad with square edges, I'm not sure it's possible to round it

**Review by Rigidity — APPROVED** [review]

**Comment by simonhyll** [comment]
@vasfvitor Yea it can be improved upon, especially considering right now I haven't touched upon how to do the splashscreen for mobile devices, but it's a start 🙂 We'll merge it and then get followup PR's for improving it

**Pull request merged**